### ISSUE TITLE
rpctest does not watch for closed processes

### ISSUE THREAD

**Issue by jrick** [decred/dcrwallet#341]
I was receiving connection timeout errors when trying to use rpctest.  After redirecting dcrd's stdout and stderr to the parent process, I noticed that it was exiting early in startup due to another issue (should be resolved after #340 is fixed).  Ideally the test harness monitors every started process for unexpected termination, and could even print a detailed message with the last few lines of stdout and stderr that the process created.

**Comment by chappjc** [comment]
It also does not check for zombie processes left over from previous runs.  I often get connection timeout errors if something went wrong in the last run (not a t.Fatal, but a crash) and a node or wallet is already bound to the TCP port it wants.  Having to kill processes with --simnet in the command is a pain and should not be necessary.

So I would suggest to go further than just making stdout and stderr reported.  Perhaps rpctest should check for leftover /tmp/rpctest-* folders, identify running processes via any .pid files, and kill them if they are the expected process name (not something else that came along and reused the pid).

**Comment by jrick** [comment]
that could be a bad idea if we try to kill them next startup, since they could have already exited for some reason, and another process was started with the same pid.  Only reliable way to do cleanup is to make sure it always happens as the test finishes.

**Comment by chappjc** [comment]
That's true, and i I agree that cleaning up on exit is the proper solution.  However it seems unlikely that there would be another wallet or node process running with the  simnet flag and same process ID.

**Comment by davecgh** [comment]
There is a much improved version of the `rpctest` package upstream in btcd [here](https://github.com/btcsuite/btcd/tree/master/rpctest) that resolves several of these issues.  Obviously there are some differences since that one was written for btcd instead of dcrwallet, but given the dcrwallet version was based on an earlier (and incomplete) version of the current btcd `rpctest` package, I would expect it's worthwhile to take a look and reconcile them.

**Comment by chappjc** [comment]
Thanks @davecgh for pointing out that version.  Hard-coding the wallet seed (https://github.com/btcsuite/btcd/blob/master/rpctest/memwallet.go#L27) is also appealing, but the differences are fairly extensive.

**Comment by jrick** [comment]
rpctest has been removed in a865b1fc923d0d21b7c57974adf83e63548c3693.
